packages/api/src/external/hie/shared.ts

```diff
@@ -0,0 +1,96 @@
+import { Coordinates } from "@metriport/core/domain/address";
+import { out } from "@metriport/core/util/log";
+import { OrgType } from "@metriport/core/domain/organization";
+import { metriportEmail as metriportEmailForCq } from "../../external/carequality/constants";
+import { metriportCompanyDetails } from "@metriport/shared";
+import { Facility } from "../../domain/medical/facility";
+import { createOrUpdateCQOrganization } from "../../external/carequality/command/cq-directory/create-or-update-cq-organization";
+import { createOrUpdateCWOrganization } from "../../external/commonwell/create-or-update-cw-organization";
+import { CqOboDetails } from "../../external/carequality/get-obo-data";
+import { buildCqOrgName } from "../../external/carequality/shared";
+import { buildCwOrgName } from "../../external/commonwell/shared";
+import { isOboFacility, isProviderFacility } from "../../domain/medical/facility";
+
+export async function createOrUpdateInCq(
+  facility: Facility,
+  cxOid: string,
+  cqOboData: CqOboDetails,
+  coordinates: Coordinates
+): Promise<void> {
+  const { log } = out("createOrUpdateInCq");
+  const isObo = isOboFacility(facility.type);
+  const isProvider = isProviderFacility(facility.type);
+  const cqOboDisabled = !isObo || !cqOboData.enabled;
+
+  if (cqOboDisabled && !isProvider) {
+    log(`CQ OBO is not enabled for this facility`);
+    return;
+  }
+
+  const cqFacilityName = cqOboData.enabled ? cqOboData.cqFacilityName : facility.data.name;
+  const cqOboOid = cqOboData.enabled ? cqOboData.cqOboOid : undefined;
+  const orgName = buildCqOrgName(cxOid, cqFacilityName, isProvider, cqOboOid);
```

Comment:
This doesn't look correct: the first parameter is not an OID, but the cx's Org name.

Previous version:

```ts
const cqOrgName = buildCqOrgName(cxOrg.name, cqFacilityName, cqOboOid);
```

That's why using an deconstructed object as parameter is good, named parameters FTW! 😅

Reply:
Yea 100% will update